tests/test_cli.py

```diff
@@ -3,54 +3,250 @@
 
 import os
 import pytest
+import re
+import subprocess
+import time
 from click.testing import CliRunner
 from dotenv import load_dotenv
 from iotedgehubdev import cli
+from iotedgehubdev import configs
+from iotedgehubdev.edgedockerclient import EdgeDockerClient
+from iotedgehubdev.hostplatform import HostPlatform
 
-filename = os.path.join(os.getcwd(), '.env')
+workingdirectory = os.getcwd()
+filename = os.path.join(workingdirectory, '.env')
 if os.path.exists(filename):
     load_dotenv(filename)
+docker_client = EdgeDockerClient()
+
 VALID_DEVICECONNECTIONSTRING = os.environ['DEVICE_CONNECTION_STRING']
+VALID_IOTHUBCONNECTIONSTRING = os.environ['IOTHUB_CONNECTION_STRING']
 
 
 @pytest.fixture
 def runner():
     return CliRunner()
 
 
+def start_process(command, is_shell):
+    process = subprocess.Popen(command, shell=is_shell,
+                               stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output, error = process.communicate()
+    if process.returncode == 0:
+        print('Process Output Message:\n' + str(output))
+    else:
+        print('Process Error Message:\n' + str(error))
+    return str(output)
+
+
+def cli_setup(runner):
+    result = runner.invoke(cli.setup, ['-c', VALID_DEVICECONNECTIONSTRING, '-g', 'iotedgetestingnow'])
+    assert 'Setup IoT Edge Simulator successfully' in result.output.strip()
```

Comment:
cli_setup is a inner method instead of test case right? If it is the case prefer to put assert into test case methods and other method just run the logic.

Reply:
> cli_setup is a inner method instead of test case right? If it is the case prefer to put assert into test case methods and other method just run the logic.

Updated code.